components/nimbus/src/lib.rs

```diff
@@ -575,6 +580,10 @@ impl NimbusClient {
         let mut event_store = self.event_store.lock().unwrap();
         event_store.record_event(event_id, None)?;
         event_store.persist_data(self.db()?)?;
+
+        // Once we have updated the EventStore, update its value on the targeting_attributes.
+        let mut state = self.mutable_state.lock().unwrap();
+        state.targeting_attributes.event_store = Some(event_store.clone());
```

Comment:
Not sure I like keeping two concurrent in memory copies. We don't need to keep the targeting attributes copy up to date, we only need to give it a current copy whenever it is consumed. At this point the event_store that is in-memory only needs to live in the targeting attributes and we don't exactly need a second one, do we?

Reply:
I've adjusted it to only set the targeting attributes value whenever it will be needed -- is there anything else I need to address for this comment?

Reply:
Nope!